lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3008,8 +3008,36 @@ void AttributeChecker::visitPropertyWrapperAttr(PropertyWrapperAttr *attr) {
 }
 
 void AttributeChecker::visitFunctionBuilderAttr(FunctionBuilderAttr *attr) {
-  // TODO: check that the type at least provides a `sequence` factory?
-  // Any other validation?
+  auto *nominal = dyn_cast<NominalTypeDecl>(D);
+  auto lookupResult = TypeChecker::lookupMember(
```

Comment:
`BuilderTransform.cpp` contains a function `builderSupports` that does the lookup for a particular function builder member; please re-use that implementation rather than implementing another lookup here, because we don't want them to get out of sync.

Reply:
Will do! It looks like that function disallows anything that isn't a `FuncDecl` (and also currently isn't used to check `buildBlock`). Should we disallow the 'weird' cases here where `buildBlock` is a `static var` with function (or callable nominal) type, or an enum case?